task.py

```diff
@@ -75,7 +76,7 @@ def create_node_port_service(self, nodeport: int) -> str:
         logger.info(f"Creating Node Port Service {out_file_yaml}")
         r = self.run_oc(f"apply -f {out_file_yaml}")
         if r.returncode != 0:
-            if "already exists" not in r.err:
+            if "already exists" not in r.err and self.reuse_node:
```

Comment:
Why do we need this switch? We aren't creating a new node port service from your task, correct?

Reply:
I think that because we are passing in the node from the iperf class the node port service is already in service, so this a way of handling that instance. The reason why we are not overriding this function it is unneccessary code duplication for one conditional.

Reply:
What is this switch doing? If the pod already exists, we will get an error, however we can ignore this since the pod we want is running.

I don't think we are setting this switch anywhere? I'm not sure what this is trying to accomplish, if we remove this now, nothing would change, no?

Reply:
The tests don't work without this switch, hence why it was added in.

Reply:
Could you elaborate why? What I don't understand is we don't call this function from this new plugin / any task other than the iperf_server task, and this currently works without these changes.

With this change if any error occurs when the iperf server is creating the node port service, it will be ignored. Is this the desired behavior?

Reply:
We pass in the node name from the iperf class and we reuse the node and it seems to hit the code path again and exit the test prematurely.

```With this change if any error occurs when the iperf server is creating the node port service, it will be ignored. Is this the desired behavior?```

This is not the case. We only set reuse_node to true when using the VF Rep class. That's the only instance it will forgo the check. Otherwise, the check is still valid.

Reply:
Lets say we are running the tests without using your plugin.

If "already exists" is not in r.err, that means a valid error has occurred, as we need to exit.

Default for self.reuse_node = False, so this will be the case when we create the iperf server instance.

Therefore, if we start an iperf server and create a node port service, and a failure occurs, it will be `if <True> and <False>` and not exit as it should. Am I thinking about this correctly?

Reply:
Why do we not need this check on create_cluster_ip_server()? And more importantly, are we creating a second iperf server somehow? We only call these functions from iperf_server.init()

Reply:
I don't understand the need for this flag... If you can provide us with a situation where it would be needed then we can better understand.

Reply:
I apologize. I hit a concurrency bug earlier and believed I needed this flag. You are correct. We do not need this flag. Removing